components/usage/pkg/apiv1/billing.go

```diff
@@ -21,24 +22,32 @@ import (
 	"gorm.io/gorm"
 )
 
-func NewBillingService(stripeClient *stripe.Client, billInstancesAfter time.Time, conn *gorm.DB) *BillingService {
+func NewBillingService(stripeClient StripeClient, billInstancesAfter time.Time, conn *gorm.DB, usageClient v1.UsageServiceClient) *BillingService {
 	return &BillingService{
 		stripeClient:       stripeClient,
 		billInstancesAfter: billInstancesAfter,
 		conn:               conn,
+		usageClient:        usageClient,
 	}
 }
 
+type StripeClient interface {
+	GetUpcomingInvoice(ctx context.Context, customerID string) (*stripe.Invoice, error)
+	UpdateUsage(ctx context.Context, creditsPerTeam map[string]map[string]float64) error
+	GetCustomerByTeamID(ctx context.Context, teamID string) (*stripesdk.Customer, error)
+	GetCustomerByUserID(ctx context.Context, userID string) (*stripesdk.Customer, error)
```

Comment:
If these are not used the "abstracted" client layer, they don't need to be listed. Here, it looks like it's only used by the mock in the tests which leads me to believe that we can drop them from here.

In Go, the interface is satisfied if all the methods in the interface are implemented. This means that you don't have to list every method on the strict in the interface (unlike with some other languages). This in turn allows you to have super thin interfaces which often only specify one method. But you can then compose interfaces. A good example is the [`io.Reader`](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/io/io.go;l=83) and [related composed interfaces](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/io/io.go;l=146)

Reply:
> it looks like it's only used by the mock in the tests

Hmm, this new interface `StripeClient` replaces the previous `*stripe.Client` and is actually used here. At least that's my understanding because when I remove them, I get something like `s.stripeClient.UpdateUsage undefined (type StripeClient has no field or method UpdateUsage)`.

Reply:
So yes, the `UpdateUsage` method is needed, but the other two I think are not needed. I'll take a look in the IDE later as well.